src/main/java/com/mraof/minestuck/inventory/musicplayer/ItemHandlerCapabilityProvider.java

```diff
@@ -0,0 +1,47 @@
+package com.mraof.minestuck.inventory.musicplayer;
+
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class ItemHandlerCapabilityProvider implements ICapabilityProvider, INBTSerializable<CompoundTag>
+{
+	private final LazyOptional<IItemHandler> lazyInitSupplierItemHandler = LazyOptional.of(this::getCachedInventory);
+	private ItemStackHandlerMusicPlayer itemStackHandlerMusicPlayer;
+	
+	@NotNull
+	@Override
+	public <T> LazyOptional<T> getCapability(@NotNull Capability<T> cap, @Nullable Direction side)
+	{
+		if(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY == cap)
+			return (lazyInitSupplierItemHandler).cast();
+		return LazyOptional.empty();
+	}
+	
+	private @NotNull ItemStackHandlerMusicPlayer getCachedInventory() {
```

Comment:
I'm wondering if it's worthwhile to go the extra step to lazily initialize the item handler (which as far as I can tell is done for performance only).
I'll have to get back to this